docs/specifications/tests/Basic-TP/basic01.md

```diff
@@ -1,199 +1,339 @@
+
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+
+## Test procedure
+
+In this section and unless otherwise specified below, the terms "[DNS Query]"
+follow the specification for DNS queries as specified in
+[DNS Query and Response Defaults]. The handling of the DNS responses on the
+DNS queries follow, unless otherwise specified below, what is specified for
+[DNS Response] in the same specification.
+
+1. If the *Child Zone* is the root zone (".") then:
+   1. Output *[B01_CHILD_FOUND]* with zone name (".").
+   2. Exit the test case.
+
+2. Create [DNS queries][DNS Query]:
+   1. Query type SOA and query name *Child Zone* ("SOA Child Query").
+   2. Query type DNAME and query name *Child Zone* ("DNAME Query").
+
+3. Create the following empty sets:
+   1. Parent name server IP, parent zone name and DNS response
+      ("Parent Name Server IP").
+   2. Parent name server IP and parent zone name ("Parent Found").
+   3. Parent name server IP and parent zone name ("Delegation Found").
+   4. Parent name server IP and parent zone name ("Non-AA Non-Delegation Found")
+   5. Parent name server IP and parent zone name ("AA NXDomain Found").
+   6. Parent name server IP and parent zone name ("AA SOA Found").
+   7. Parent name server IP and parent zone name ("AA CNAME Found").
+   8. Parent name server IP, parent zone name and DNAME target
+      ("AA DNAME Found").
+   9. Parent name server IP and parent zone name ("AA NODATA Found").
+   10. Name server IP and zone name ("Remaining Servers").
+   11. Name server IP ("Handled Servers").
+
+> *Note that a referral is a DNS response with [RCODE Name] NoError, AA flag
+> unset and NS records in the authority section. The answer section is empty
+> or with CNAME record or records. If the query type is CNAME, then the answer
+> section must be empty. The additional section may contain address records
+> (A and AAAA) for the name server names from the NS (glue records).*
+
+4. Insert all addresses from *Root Name Servers* and the root zone name into the
+   *Remaining Servers* set.
+5. While the *Remaining Servers* is non-empty pick next name server IP address
+   and zone name from the set ("Server Address" and "Zone Name") and do:
+   1. Remove the IP address from *Remaining Servers*.
+   2. Insert *Server Address* into *Handled Servers*
+   3. [Send](#terminology) *SOA Child Query* to *Server Address*.
+   4. If *Server Address* does not respond at all, with an invalid DNS response
+      or with an [RCODE Name] besides NoError and NXDomain, then ignore it.
```

Comment:
Or rather, what is a valid DNS response? -- This is a general question applicable for all test cases sending queries. -- Instead of trying to resolve this now, I have removed it, and we can try to define it later.

Reply:
Sounds like a good idea. Maybe in the in the Query And Response Defaults document? Yeah, I know, it was intended to be about defaults. But I think it would make a lot of sense to broaden its scope to cover the entire handling of a single DNS request-response cycle.

Reply:
Yes, maybe in a broadened document.

Reply:
I created #1096 so we don't forget this.